### DISCUSSION TITLE
Porting this repo to soccer

### DISCUSSION THREAD

**Discussion by AndSalaMove** [slxmmy999/InfiniteBaseballGrid#32] [Q&A]
Hey,
first of all thanks for sharing your code.

I would like to make an equivalent version of this game but restricted to italian soccer players.
Could you please explain what is there to change? I would like to keep your backend engine and simply change the db (still have to figure out from where) and the teams/logos.

Thanks!

**Comment by AndSalaMove** [comment]
PS I have solid experience with python, but close to none with JS

**Reply by slxmmy999** [reply]
I'd love to help you port this to soccer! The backend does need to be changed because you will need to fetch player team info and stats data for soccer instead of baseball, however the good news is the backend is in Python and you should be able to do that pretty easily if you have experience. Would you be interested in adding a soccer mode to the current game or do you want to host your own separate version? I'm happy to work with you as much as you need on this.

**Reply by AndSalaMove** [reply]
Hey, thanks for your answer. 
We could surely extend the project to soccer and place it on your website, it would be cool (There is already the soccer version of the grid quiz [here](https://www.crossovergrid.com/soccer) and [here](https://www.immaculatefooty.com/).
Moreover, I was looking to build a local version of the quiz using only players from the Italian Serie A, to be played with my friends to joke around. Anyway, putting this version online would be equally cool (but I don't know how much interest it would gather over time)

**Reply by slxmmy999** [reply]
Honestly I'm not concerned with the interest it gathers, I am just happy to have interest in expending the game. Do you know what API you can use to get the soccer data from?

**Reply by AndSalaMove** [reply]
Not yet, what are you using for baseball? 
I know FBRef and maybe trasfermarkt might have the data we are looking for, but i'm not aware of any free APIs at the moment, we'll have to look it up.

**Reply by slxmmy999** [reply]
I'm using the official MLB stats API. I may also end up using some BBRef data too in the future.

**Reply by AndSalaMove** [reply]
How does it work for you? Are you making a call to the MLB database every time a player is selected in the game, or are you hosting the data on a separate local db?
PS I have a full time job + I'm moving houses so sorry if I'm not doing anything...

**Reply by slxmmy999** [reply]
I don't have the database storage to cache all of the player combinations in my DB, so I query the MLB API every time a player is selected to check their teams and get their image and name if they work. I do however have my own database that I store the statistics for how often players are selected for each combination in order to calculate the rarity scores, which is updated every time the player is selected and is correct. Also I just moved and I'm working + doing school so I totally get it I'm super busy too there's no pressure at all.

**Reply by slxmmy999** [reply]
I don't have the database storage to cache all of the player combinations in my DB, so I query the MLB API every time a player is selected to check their teams and get their image and name if they work. I do however have my own database that I store the statistics for how often players are selected for each combination in order to calculate the rarity scores, which is updated every time the player is selected and is correct. Also I just moved and I'm working + doing school so I totally get it I'm super busy too there's no pressure at all.
